### PR TITLE
Add ListExtension Method SelectWhereNot

### PR DESCRIPTION
**Pull request by TheDioBrando**
Add Func funcCatColorIsDark
Add Func funcPersonPhoneNumberStartsWith7

### CODE REVIEW

File: src/FuncHW/ListExtensions.cs [hunk 1]
```diff
@@ -12,5 +12,18 @@ public static class ListExtensions
         //add extension for List CountElements: receives Func, returns the number of elements that match the condition
 
         //add extension for List SelectWhereNot: receives Func, returns the list of elements that don't match the condition
+        public static List<T> SelectWhereNot<T>(this List<T> values, Func<T, bool> func)
+            where T : class
```
Comment (TIove) [inline src/FuncHW/ListExtensions.cs]: For what this condition? What if I want to use this extension with 'int' or other struct?

File: src/FuncHW/ListExtensions.cs [hunk 2]
```diff
@@ -12,5 +12,18 @@ public static class ListExtensions
...
+        {
+            List<T> resultList = new List<T>();
+
+            foreach (var item in values)
+            {
+                if(!func(item))
```
Comment (TIove) [inline src/FuncHW/ListExtensions.cs]: Space after if. Should be `if ()`.

File: src/FuncHW/Program.cs [hunk 3]
```diff
@@ -27,11 +27,18 @@ static void Main()
 
             //Func*** funcPersonIsActive = ***;
             //Func*** funcPersonPhoneNumberStartsWith7 = ***;
+            Func<Person, bool> funcPersonPhoneNumberStartsWith7 = (Person person) => 
+                person.PhoneNumber / 10000 == 7;
```
Comment (TIove) [inline src/FuncHW/Program.cs]: No, that's bad to use `/ 10000`. What if appear new phone number with six symbols? e.x. 123456. 
Try to come up with something else.

File: src/FuncHW/Program.cs [hunk 4]
```diff
@@ -27,11 +27,18 @@ static void Main()
...
+
             //Func*** funcPersonIsChild = ***;
             //Func*** funcPersonHasShortName = ***;
 
             //Func*** funcCatIsDomestic = ***;
             //Func*** funcCatColorIsDark = ***;
+            Func<Cat, bool> funcCatColorIsDark=(Cat cat)=>
```
Comment (TIove) [inline src/FuncHW/Program.cs]: Be careful with spaces. `...Dark = (Cat cat) =>`

File: src/FuncHW/Program.cs [hunk 5]
```diff
@@ -27,11 +27,18 @@ static void Main()
...
+            Func<Person, bool> funcPersonPhoneNumberStartsWith7 = (Person person) => 
+                person.PhoneNumber / 10000 == 7;
+
             //Func*** funcPersonIsChild = ***;
             //Func*** funcPersonHasShortName = ***;
 
             //Func*** funcCatIsDomestic = ***;
             //Func*** funcCatColorIsDark = ***;
+            Func<Cat, bool> funcCatColorIsDark=(Cat cat)=> 
+                string.Equals(cat.Color, "grey",StringComparison.OrdinalIgnoreCase)
```
Comment (TIove) [inline src/FuncHW/Program.cs]: Add space after comma.

File: src/FuncHW/Program.cs [hunk 6]
```diff
@@ -27,11 +27,18 @@ static void Main()
...
             //Func*** funcPersonHasShortName = ***;
 
             //Func*** funcCatIsDomestic = ***;
             //Func*** funcCatColorIsDark = ***;
+            Func<Cat, bool> funcCatColorIsDark=(Cat cat)=> 
+                string.Equals(cat.Color, "grey",StringComparison.OrdinalIgnoreCase) 
+                || string.Equals(cat.Color, "black", StringComparison.OrdinalIgnoreCase);
+
             //Func*** funcCatNameConteinsU = ***;
             //Func*** funcCatIsDomesticAndWhite = ***;
```
Comment (TIove) [inline src/FuncHW/Program.cs:55]: //write result to variable

Where is your result? Please, add new variable with extension calling.

File: src/FuncHW/ListExtensions.cs (hunk 2 above)
Comment (TIove) [reply src/FuncHW/ListExtensions.cs]: Also, that's not so bad imho, but in DigitalOffice we write always with brackets `{ }`

File: src/FuncHW/Program.cs (hunk 6 above)
Comment (TheDioBrando) [reply src/FuncHW/Program.cs:55]: Done

File: src/FuncHW/Program.cs (hunk 5 above)
Comment (TheDioBrando) [reply src/FuncHW/Program.cs]: Done

File: src/FuncHW/Program.cs (hunk 4 above)
Comment (TheDioBrando) [reply src/FuncHW/Program.cs]: Done

File: src/FuncHW/ListExtensions.cs (hunk 2 above)
Comment (TheDioBrando) [reply src/FuncHW/ListExtensions.cs]: Done

File: src/FuncHW/ListExtensions.cs (hunk 1 above)
Comment (TheDioBrando) [reply src/FuncHW/ListExtensions.cs]: Delete this condition

File: src/FuncHW/Program.cs (hunk 3 above)
Comment (TheDioBrando) [reply src/FuncHW/Program.cs]: Remake check for any length of number.

**Review by vorobevamv — APPROVED** [review]

File: src/FuncHW/Program.cs [hunk 7]
```diff
@@ -25,18 +25,25 @@ static void Main()
 
             List<Cat> cats = new() { barsik, pushok, murzik, dymok, murka };
 
-            //Func*** funcPersonIsActive = ***;
```
Comment (TIove) [inline src/FuncHW/Program.cs:28]: Rollback this comments, please

You can delete only commented-functions that you implemented

Comment (TIove) [reply src/FuncHW/Program.cs:28]: This is necessary, because, when your merge request will be merged, all other students will not see theirs tasks

**Review by TIove — APPROVED** [review]

File: src/FuncHW/Program.cs (hunk 7 above)
Comment (TheDioBrando) [reply src/FuncHW/Program.cs:28]: Sorry, returned comments

**Review by liza459 — APPROVED** [review]

**Review by MorozVl — COMMENTED** [review]
LGTM

**Review by MorozVl — APPROVED** [review]

**Pull request merged**